### DISCUSSION TITLE
The C# code converted from curl is not working

### DISCUSSION THREAD

**Discussion by testkongqi** [olsh/curl-to-csharp#73] [General]
Your tool is great! it helps me to get restful api call responses with C# code from curl command. But the C# code from the following curl command doesn't work. Could you help me out? thanks
curl -X POST -H "Content-Type: application/x-www-form-urlencoded" --referer https://yahoo.com/api/token -H "X-CSRFToken: dfdfffff" --data "username=dummy&password=header" --cookie "mytoken=dfgdsff" https://yahoo.com/api/token -k -D --o /dev/null

**Comment by olsh** [comment]
Hi @testkongqi

Well, this is probably because `--referer` parameter is not supported in the converter.
Which error did you get?

**Comment by olsh** [comment]
Just added `--referer` support. You can try again.

**Comment by testkongqi** [comment]
403 forbidden.

thanks for quick response.

**Comment by testkongqi** [comment]
I tried added the Referer as a header property and get 404 not found.

**Comment by testkongqi** [comment]
Hi there,
another question. when i tried to convert the following curl command, I got
statuscode 401 unauthorized
curl --location --request GET "https://yahoo.com/api/token?id=3343"
--header "Authorization: Basic tggdfdf".

could you take a look?

Thanks,

**Comment by olsh** [comment]
It's pretty hard to say what's wrong with your request. 
The curl requests convert without any issues.
